CMakeLists.txt

```diff
@@ -12,19 +12,21 @@ project(diagnostics)
 # include shared compiler configs
 include(eng/native/configurepaths.cmake)
 include(${CLR_ENG_NATIVE_DIR}/configurecompiler.cmake)
+set(CLR_SHARED_DIR ${CLR_REPO_ROOT_DIR}/src/shared)
 
-# override some global configs
-if (MSVC)
-  add_compile_options(/wd4960 /wd4961 /wd4603 /wd4627 /wd4838 /wd4456 /wd4457 /wd4458 /wd4459 /wd4091 /we4640)
-  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:/EHsc>)
-  add_compile_options($<$<OR:$<CONFIG:Release>,$<CONFIG:Relwithdebinfo>>:/MT>)
-  add_compile_options($<$<OR:$<CONFIG:Debug>,$<CONFIG:Checked>>:/MTd>)
-else (MSVC)
-  add_compile_options(-Wno-unused-parameter)
-  add_compile_options(-Wno-unused-variable)
-  add_compile_options(-Wno-implicit-fallthrough)
+if(CLR_CMAKE_HOST_UNIX)
+  # The -fms-extensions enable the stuff like __if_exists, __declspec(uuid()), etc.
+  add_compile_options(-fms-extensions)
+  #-fms-compatibility      Enable full Microsoft Visual C++ compatibility
+  #-fms-extensions         Accept some non-standard constructs supported by the Microsoft compiler
   add_compile_options(-fvisibility=default)
-endif (MSVC)
+endif(CLR_CMAKE_HOST_UNIX)
+
+if (CMAKE_VERSION VERSION_LESS "3.16")
```

Comment:
Yes for __uuidof/__declspec(uuid(x)) on Linux/MacOS. The runtime doesn't need it anymore because all the __uuidof are midl generated h files that have a IID_<class> defined automatically generated.  For the SOS COM classes like debuggerservices.h/host.h/runtime.h, etc. need these extensions.

Reply:
This is also the reason diagnostics\src\shared\pal\inc\rt\rpc.h is different from the runtime.